pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/ParallelWheelGenerationTask.java

```diff
@@ -0,0 +1,166 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.tasks;
+
+import com.linkedin.gradle.python.PythonExtension;
+import com.linkedin.gradle.python.extension.PlatformTag;
+import com.linkedin.gradle.python.extension.PythonDetails;
+import com.linkedin.gradle.python.extension.PythonTag;
+import com.linkedin.gradle.python.util.PackageInfo;
+import com.linkedin.gradle.python.util.internal.TaskTimer;
+import com.linkedin.gradle.python.wheel.WheelCache;
+import org.apache.commons.io.FileUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.internal.logging.progress.ProgressLogger;
+import org.gradle.internal.logging.progress.ProgressLoggerFactory;
+import org.gradle.process.ExecResult;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ParallelWheelGenerationTask extends DefaultTask {
+
+    private static final Logger logger = Logging.getLogger(ParallelWheelGenerationTask.class);
+
+    private FileCollection filesToConvert;
+    private File cacheDir;
+    private PythonExtension extension;
+    private PythonDetails pythonDetails;
+    private AtomicInteger counter = new AtomicInteger();
+
+    @TaskAction
+    public void buildWheels() {
+
+        ProgressLoggerFactory progressLoggerFactory = getServices().get(ProgressLoggerFactory.class);
+        ProgressLogger progressLogger = progressLoggerFactory.newOperation(ParallelWheelGenerationTask.class);
+        progressLogger.setDescription("Building wheels");
+
+        PythonTag pythonTag = PythonTag.findTag(getProject(), getPythonDetails());
+        PlatformTag platformTag = PlatformTag.makePlatformTag(getProject(), getPythonDetails());
+
+        WheelCache wheelCache = new WheelCache(cacheDir, pythonTag, platformTag);
+
+        progressLogger.started();
+
+        TaskTimer taskTimer = new TaskTimer();
+
+        Set<File> files = getFilesToConvert().getFiles();
+        int totalSize = files.size();
+        files.parallelStream().forEach(file -> {
+            PackageInfo packageInfo = PackageInfo.fromPath(file.getPath());
+            TaskTimer.TickingClock clock = taskTimer.start(packageInfo.getName() + "-" + packageInfo.getVersion());
+            makeWheelFromSdist(progressLogger, totalSize, wheelCache, file);
+            clock.stop();
+        });
+
+        try {
+            FileUtils.write(getBuildReport(), taskTimer.buildReport());
+        } catch (IOException ignore) {
+        }
```

Comment:
Can you add a comment inside this block, such as `// PASS`
Similar for other instances of this pattern.